pkgs/development/libraries/science/math/magma/generic.nix

```diff
@@ -164,7 +161,11 @@ stdenv.mkDerivation {
     homepage = "http://icl.cs.utk.edu/magma/index.html";
     platforms = platforms.unix;
     maintainers = with maintainers; [ connorbaker ];
-    # CUDA and ROCm are mutually exclusive
-    broken = cudaSupport && rocmSupport || cudaSupport && strings.versionOlder cudaVersion "9";
+
+    # Cf. https://bitbucket.org/icl/magma/src/fcfe5aa61c1a4c664b36a73ebabbdbab82765e9f/CMakeLists.txt#lines-20
+    broken =
+      !(cudaSupport || rocmSupport) # At least one back-end enabled
+      || (cudaSupport && rocmSupport) # Mutually exclusive
```

Comment:
See previous comment.
We can get rid of this with ``gpuBackend`` or something to that flavor.

Reply:
I'd actually much rather keep `broken` because I don't feel sufficiently familiar with magma to attempt constraining the user in what they can try and achieve with overrides. With `broken` we just signal we expect certain configurations of inputs to fail everything else being constant